lib/logger/lib/logger.ex

```diff
@@ -76,45 +76,38 @@ defmodule Logger do
     * `:backends` - the backends to be used. Defaults to `[:console]`.
       See the "Backends" section for more information.
 
-    * `:compile_time_purge_level` - purges *at compilation time* all calls that
-      have log level lower than the value of this option. This means that
-      `Logger` calls with level lower than this option will be completely
-      removed at compile time, accruing no overhead at runtime. Defaults to
-      `:debug` and only applies to the `Logger.debug/2`, `Logger.info/2`,
-      `Logger.warn/2`, and `Logger.error/2` macros (for example, it doesn't apply to
-      `Logger.log/3`). Note that arguments passed to `Logger` calls that are
-      removed from the AST at compilation time are never evaluated, thus any
-      function call that occurs in these arguments is never executed. As a
-      consequence, avoid code that looks like `Logger.debug("Cleanup:
-      #{perform_cleanup()}")` as in the example `perform_cleanup/0` won't be
-      executed if the `:compile_time_purge_level` is `:info` or higher.
-
     * `:compile_time_purge_matching` - purges *at compilation time* all calls
-      that match the given metadata. This configuration expects a list of
+      that match the given conditions. This means that `Logger` calls with
+      level lower than this option will be completely removed at compile time,
+      accruing no overhead at runtime. This configuration expects a list of
       keyword lists. Each keyword list contains a metadata key and the matching
-      value that should be purged. Remember that if you want to purge log calls
-      from a dependency, the dependency must be recompiled.
+      value that should be purged. A special key named `:level_lower_than` can
+      be used to purge all messages with a lower logger level. Remember that
+      if you want to purge log calls from a dependency, the dependency must be
+      recompiled.
 
     * `:compile_time_application` - sets the `:application` metadata value
       to the configured value at compilation time. This configuration is
       usually only useful for build tools to automatically add the
       application to the metadata for `Logger.debug/2`, `Logger.info/2`, etc.
       style of calls.
 
-  For example, to configure the `:backends` and `:compile_time_purge_level`
-  options in a `config/config.exs` file:
+  For example, to configure the `:backends` and purge all calls that happen
```

Comment:
I would remove "For example" because reading it sounds like it follows from the previous sentence which is an unrelated to config. We seem to list a few examples here, so I would suggest adding an "### Examples" heading.